selfdrive/locationd/torqued.py

```diff
@@ -0,0 +1,208 @@
+        active = bool(np.interp(t, np.array(self.raw_points_t['active']) + self.lag, self.raw_points['active']))
+        steer_override = bool(np.interp(t, np.array(self.raw_points_t['steer_override']) + self.lag, self.raw_points['steer_override']))
+        vego = np.interp(t, np.array(self.raw_points_t['vego']) + self.lag, self.raw_points['vego'])
+        steer = np.interp(t, np.array(self.raw_points_t['steer_torque']) + self.lag, self.raw_points['steer_torque'])
+        if active and (not steer_override) and (vego > MIN_VEL) and (abs(steer) > STEER_MIN_THRESHOLD):
+          lateral_acc = (vego * yaw_rate) - (np.sin(roll) * ACCELERATION_DUE_TO_GRAVITY)
+          self.filtered_points.add_point(steer, lateral_acc)
+
+
+def main(sm=None, pm=None):
+  set_realtime_priority(1)
+
+  if sm is None:
+    sm = messaging.SubMaster(['carControl', 'carState', 'liveLocationKalman'], poll=['liveLocationKalman'])
+
+  if pm is None:
+    pm = messaging.PubMaster(['liveTorqueParameters'])
+
+  params_reader = Params()
+  CP = car.CarParams.from_bytes(params_reader.get("CarParams", block=True))
+  params = params_reader.get("LiveTorqueParameters")
+  estimator = TorqueEstimator(CP, params)
+
+  while True:
+    sm.update()
+    if sm.all_checks():
+      for which in sm.updated.keys():
+        if sm.updated[which]:
+          t = sm.logMonoTime[which] * 1e-9
+          estimator.handle_log(t, which, sm[which])
+
+    if sm.updated['liveLocationKalman']:
+      # print(sm.frame, [len(v) for v in estimator.filtered_points.buckets.values()])
+      msg = messaging.new_message('liveTorqueParameters')
+      msg.valid = sm.all_checks()
+      liveTorqueParameters = msg.liveTorqueParameters
+
+      if estimator.filtered_points.is_valid():
+        try:
+          slope, offset, friction_coeff = estimator.estimate_params()
+          # print(slope, offset, friction_coeff)
+        except Exception as e:
+          # print(e)
+          slope = offset = friction_coeff = None
+          cloudlog.exception(f"Error computing live torque params: {e}")
+
+        if estimator.is_sane(slope, offset, friction_coeff):
+          liveTorqueParameters.liveValid = True
+          liveTorqueParameters.slopeRaw = slope
+          liveTorqueParameters.offsetRaw = offset
+          liveTorqueParameters.frictionCoefficientRaw = friction_coeff
+          estimator.slopeFiltered.update(slope)
+          estimator.offsetFiltered.update(offset)
+          estimator.frictionCoefficientFiltered.update(friction_coeff)
+        else:
+          cloudlog.exception("live torque params are numerically unstable")
+          liveTorqueParameters.liveValid = False
+          # estimator.reset()
+      else:
+        liveTorqueParameters.liveValid = False
+      liveTorqueParameters.slopeFiltered = estimator.slopeFiltered.x
+      liveTorqueParameters.offsetFiltered = estimator.offsetFiltered.x
+      liveTorqueParameters.frictionCoefficientFiltered = estimator.frictionCoefficientFiltered.x
```

Comment:
```suggestion
      liveTorqueParameters.slopeFiltered = float(estimator.slopeFiltered.x)
      liveTorqueParameters.offsetFiltered = float(estimator.offsetFiltered.x)
      liveTorqueParameters.frictionCoefficientFiltered = float(estimator.frictionCoefficientFiltered.x)
```

Kept crashing for me